### DISCUSSION TITLE
Unable to find destination for [0x2ac8ba92c0:0x2ac8ba9300] on system.l2buses0

### DISCUSSION THREAD

**Discussion by qqli579** [gem5/gem5#405] [Q&A] [answered]
hi，all：
       This error occurred when I used the fs_linux.py script to perform full system simulation:
src/cpu/o3/fetch.cc:604: warn: Address 0x3fa7e62b00 is outside of physical memory, stopping fetch
src/cpu/o3/fetch.cc:604: warn: Address 0x3fa7e62b00 is outside of physical memory, stopping fetch
src/mem/xbar.cc:358: fatal: Unable to find destination for [0x2ac8ba92c0:0x2ac8ba9300] on system.l2buses0
     The command I ran is as follows:
./build/RISCV/gem5.fast ./configs/example/riscv/fs_linux.py --fast-forward=10000000000 --cpu-type=RiscvO3CPU --disk-image=./image/rootfs2.img --kernel=./image/kernel.elf --num-cpus=8 --num-clusters=2 --cpu-clock=3GHz --sys
-clock=1.5GHz --mem-type=LPDDR3_1600_1x32 --mem-size=4GB --caches --l2cache --l3cache
 --cacheline_size=64 --l1d_size=64kB --l1i_size=64kB --l1d_assoc=8 --l1i_assoc=8 --l2shared --l2_banks
=8 --l2_size=512kB --l2_assoc=16 --l3_size=4MB --l3_assoc=16 --l3_banks=8
      The rootfs2.img and kernel files are made by myself. According to the error message, this address is 171GB, which exceeds the 4GB I configured. 
     I read that some people on the Internet said that you can increase the memory size or use dummy port to avoid this problem. But there is no other way to solve this problem or how to use dummy port to solve this problem?
   thanks

**Comment by Harshil2107** [comment, marked as answer]
Hi,

The `fs_linux.py` is an example script and is not well-tested and managed.

Increasing the memory size could potentially solve this problem, but it might be hiding an underlying issue. It is possible that there is a bug in the simulator or in the binary that is causing these addresses to be computed incorrectly.

You might want to try the standard riscv board in the standard library.
To use local resources, you would also need to create the (https://github.com/gem5/gem5/blob/48a40cf2f5182a82de360b7efa497d82e06b1631/src/python/gem5/resources/resource.py#L264)  and DiskImageResource(https://github.com/gem5/gem5/blob/48a40cf2f5182a82de360b7efa497d82e06b1631/src/python/gem5/resources/resource.py#L168C7-L168C24) objects 

Here is a barebones example of setting up the RISCV board.

```
rom gem5.components.boards.riscv_board import RiscvBoard
from gem5.components.memory import SingleChannelDDR3_1600
from gem5.components.processors.simple_processor import SimpleProcessor
from gem5.components.cachehierarchies.classic.private_l1_private_l2_cache_hierarchy import (
    PrivateL1PrivateL2CacheHierarchy,
)
from gem5.components.processors.cpu_types import CPUTypes
from gem5.isas import ISA
from gem5.utils.requires import requires
from gem5.resources.resource import Resource
from gem5.simulate.simulator import Simulator

chache_hierarchy = PrivateL1PrivateL2CacheHierarchy(
    l1d_size="64KiB", l1i_size="64KiB", l2_size="512KiB"
)

memory = SingleChannelDDR3_1600(size= 4GB)

processor = SimpleProcessor(
	cpu_type= CPUTypes.O3, isa = ISA.RISCV
)

board = RiscvBoard(
	clk_freq= "3GHz",
	processor = processor",
	memory = memory,
	chache_hierarchy = cache_hierarchy,
)

board.set_kernel_disk_workload(
	kernel= KernelResoruce(local_path = [path to kernel]),
	disk_image= DiskImageResource(local_path = [path to disk image]),
)

sim = simulator(board = board)
sim.run()
```
